modules/passkey/test/userStories/executeUserOpFromPasskeySigner.ts

```diff
@@ -0,0 +1,224 @@
+      entryPoint,
+      singleton,
+      signerFactory,
+      navigator,
+      verifier,
+      SafeL2,
+      signer,
+      credential,
+    }
+  })
+
+  it('should execute a userOp with WebAuthn signer as owner', async () => {
+    // Step 1: Setup the contracts
+    const { user, proxyFactory, safeModuleSetup, module, entryPoint, singleton, navigator, SafeL2, signer, credential } = await setupTests()
+
+    // Step 2: Create a userOp, sign it with Passkey signer.
+    const safeSalt = Date.now()
+    const initializer = safeModuleSetup.interface.encodeFunctionData('enableModules', [[module.target]])
+
+    const setupData = singleton.interface.encodeFunctionData('setup', [
+      [signer],
+      1n,
+      await safeModuleSetup.getAddress(),
+      initializer,
+      await module.getAddress(),
+      ethers.ZeroAddress,
+      0,
+      ethers.ZeroAddress,
+    ])
+
+    // Predict the Safe address to construct the userOp
+    const safe = await proxyFactory.createProxyWithNonce.staticCall(singleton, setupData, safeSalt)
+
+    const deployData = proxyFactory.interface.encodeFunctionData('createProxyWithNonce', [singleton.target, setupData, safeSalt])
+
+    // Create safeOp
+    const safeOp = buildSafeUserOpTransaction(
+      safe,
+      user.address,
+      ethers.parseEther('0.0'),
+      '0x',
+      await entryPoint.getNonce(safe, 0),
+      await entryPoint.getAddress(),
+      false,
+      true,
+      {
+        initCode: ethers.solidityPacked(['address', 'bytes'], [proxyFactory.target, deployData]),
+        verificationGasLimit: 700000,
+      },
+    )
+
+    const packedUserOp = buildPackedUserOperationFromSafeUserOperation({
+      safeOp,
+      signature: '0x',
+    })
+
+    const opHash = await module.getOperationHash(packedUserOp)
+
+    const assertion = navigator.credentials.get({
+      publicKey: {
+        challenge: ethers.getBytes(opHash),
+        rpId: 'safe.global',
+        allowCredentials: [{ type: 'public-key', id: new Uint8Array(credential.rawId) }],
+        userVerification: 'required',
+      },
+    })
+
+    // Build contract signature that the Safe will forward to the signer contract
+    const signature = buildSignatureBytes([
+      {
+        signer: signer as string,
+        data: encodeWebAuthnSignature(assertion.response),
+        dynamic: true,
+      },
+    ])
+
+    // Send 1 ETH to the Safe
+    await user.sendTransaction({ to: safe, value: ethers.parseEther('1') }).then((tx) => tx.wait())
+    // Check if Safe is not already created
+    expect(await ethers.provider.getCode(safe)).to.equal('0x')
+
+    // Step 3: Execute the userOp that deploys a safe with passkey signer as owner.
+    await (
+      await entryPoint.handleOps(
+        [
+          {
+            ...packedUserOp,
+            signature: ethers.solidityPacked(['uint48', 'uint48', 'bytes'], [safeOp.validAfter, safeOp.validUntil, signature]),
+          },
+        ],
+        user.address,
+      )
+    ).wait()
```

Comment:
Same comment as #343 (pre-req to user story).

Reply:
done: [2093cd0](https://github.com/safe-global/safe-modules/pull/344/commits/2093cd0ef983b45eb2877a62f2e304b86e950a7b) [af0d2e7](https://github.com/safe-global/safe-modules/pull/344/commits/af0d2e758112d1d12baa2b2235fead3e997c5503)